src/utils/saml.ts

```diff
@@ -670,18 +672,103 @@ export const getPreValidateResponse = (
+      );
+      const AssertionIssueInstantDelta = String(
+        startTime - info.AssertionIssueInstant.getTime()
+      );
+      const AssertionNotBeforeDelta = pipe(
+        info.Assertion.getAttribute("NotBefore"),
+        NonEmptyString.decode,
+        E.chain(UTCISODateFromString.decode),
+        E.map(NotBefore => String(startTime - NotBefore.getTime())),
+        E.getOrElseW(() => InfoNotAvailable)
+      );
+      const AssertionSubjectNotOnOrAfterDelta = pipe(
+        info.Assertion.getElementsByTagNameNS(
+          SAML_NAMESPACE.ASSERTION,
+          "Subject"
+        ).item(0),
+        O.fromNullable,
+        O.chainNullableK(Subject =>
+          Subject.getElementsByTagNameNS(
+            SAML_NAMESPACE.ASSERTION,
+            "SubjectConfirmation"
+          ).item(0)
+        ),
+        O.chainNullableK(SubjectConfirmation =>
+          SubjectConfirmation.getElementsByTagNameNS(
+            SAML_NAMESPACE.ASSERTION,
+            "SubjectConfirmationData"
+          ).item(0)
+        ),
+        E.fromOption(() => new Error("Could not find elements")),
+        E.chainW(extractNotOnOrAfterDelta),
+        E.getOrElseW(() => InfoNotAvailable)
+      );
+      const AssertionConditionsNotOnOrAfterDelta = pipe(
+        info.Assertion.getElementsByTagNameNS(
+          SAML_NAMESPACE.ASSERTION,
+          "Conditions"
+        ).item(0),
+        O.fromNullable,
+        E.fromOption(() => new Error("Could not find elements")),
+        E.chainW(extractNotOnOrAfterDelta),
+        E.getOrElseW(() => InfoNotAvailable)
+      );
+
+      const timings = {
+        AssertionConditionsNotOnOrAfterDelta,
+        AssertionIssueInstantDelta,
+        AssertionNotBeforeDelta,
+        AssertionSubjectNotOnOrAfterDelta,
+        ResponseIssueInstantDelta
+      };
+
+      eventHandler({
+        data: {
+          idpIssuer,
+          message: "Deltas infos to determine a valid clockSkewMs",
+          requestId,
+          ...timings
+        },
+        name: "spid.info.deltas",
+        type: "INFO"
+      });
+    }
+
+    return TE.right(void 0);
+  };
+
   pipe(
     responseElementValidationStep,
     TE.chain(returnRequestAndResponseStep),
     TE.chain(parseSAMLRequestStep),
     TE.chain(getIssueInstantFromRequestStep),
-    TE.chain(issueInstantValidationStep),
-    TE.chain(assertionIssueInstantValidationStep),
+    TE.chainFirst(issueInstantValidationStep),
+    TE.chainFirst(assertionIssueInstantValidationStep),
     TE.chain(authnContextClassRefValidationStep),
-    TE.chain(attributesValidationStep),
-    TE.chain(responseIssuerValidationStep),
-    TE.chain(assertionIssuerValidationStep),
-    TE.chainW(transformValidationStep),
+    TE.chainFirst(attributesValidationStep),
+    TE.chainFirst(responseIssuerValidationStep),
+    TE.chainFirst(assertionIssuerValidationStep),
+    TE.chainFirstW(transformValidationStep),
+    // log timings infos
+    TE.chainFirstW(extractAndLogTimings),
```

Comment:
Can we make this step configurable like `strictValidationOptions`?
The caller (aka `io-backend`) can decide to enable or not the collection of timings events.

Reply:
i added in https://github.com/pagopa/io-spid-commons/pull/121/commits/38b94c3f8e8abe80e91d40c449de14b28faff164 a `hasDeltasLogging` boolean as an optional feature flag.